tools/clang/unittests/HLSL/DxilContainerTest.cpp

```diff
@@ -419,7 +422,17 @@ class DxilContainerTest : public ::testing::Test {
     if (hrVer == E_NOINTERFACE) return false;
     VERIFY_SUCCEEDED(hrVer);
     VERIFY_SUCCEEDED(pVersionInfo->GetVersion(&Major, &Minor));
-    return Major == 1 && (Minor >= 1);
+    return !(Major == 1 && Minor < 1);
+  }
+
+  bool DoesValidatorSupportShaderHash() {
+    CComPtr<IDxcVersionInfo> pVersionInfo;
+    UINT Major, Minor;
+    HRESULT hrVer = m_dllSupport.CreateInstance(CLSID_DxcValidator, &pVersionInfo);
```

Comment:
This assumes things about the vtable layout of the object implementing IDxcValidator. I've seen it elsewhere so I guess it works, but I'd rather we played by the rules and CreateInstance an IDxcValidator and then QI IDxcVersionInfo.

Reply:
Actually CreateInstance does the QI based on the passed in interface, doesn't it?

Reply:
Yeah, each Create* implementation for each object type returns result of QI on interface.